source/app/controllers/student_dashboard_controller.rb

```diff
@@ -47,6 +47,42 @@ def submit_evaluation
     end
   end
 
+  def update_settings
+    # Separate logic for password update to prevent logout before all settings are updated
```

Comment:
Since this method is the same as update_settings for the instructor, maybe place this method inside a new SettingsUpdateService under `app/services/settings_update_service.rb`. You can then call this method here:
```ruby
def update_settings
  SettingsUpdateService.update(current_user, params[:settings])
end
```